actix-http/src/encoding/decoder.rs

```diff
@@ -45,6 +46,10 @@ where
             ContentEncoding::Gzip => Some(ContentDecoder::Gzip(Box::new(
                 GzDecoder::new(Writer::new()),
             ))),
+            ContentEncoding::Zstd => match ZstdDecoder::new(Writer::new()) {
+                Ok(decoder) => Some(ContentDecoder::Zstd(Box::new(decoder))),
+                Err(_) => None,
```

Comment:
Is this OK to ignore an error here?
What happens if the `zstd` decoder was not created and then `Decoder` is used to decode `zstd`-compressed data?

Reply:
This is a good question. Actually my first implementation was to just unwrap the `ZstdDecoder::new` result.
Looking at the code I do not really see how this can be an `Err`:
- [stream::write::Decoder::new](https://docs.rs/zstd/0.8.2+zstd.1.5.0/src/zstd/stream/write/mod.rs.html#245-247)
- calls [stream::write::Decoder::with_dictionary](https://docs.rs/zstd/0.8.2+zstd.1.5.0/src/zstd/stream/write/mod.rs.html#253-257)
- calls [raw::Decoder::with_dictionary](https://docs.rs/zstd/0.8.2+zstd.1.5.0/src/zstd/stream/raw.rs.html#142-149)
- which panic if dictionary is invalid. But in our case, I do not understand how the default dictionary can be invalid.

I always "fear" unwrapping and prefer handle error case (even if they are not really possible) in case of library changes.
But indeed in this case, behavior will be "strange".
So it is up to you, I can update this to just unwrap the `ZstdDecoder::new` result or keep this as it.

Reply:
I'd go with `expect`. It *shouldn't* panic, thus it must be safe in theory, but just for the case it will *ever* panic, we should provide a helpful error message telling user that it's a bug in either `actix-web` or `zstd` and this issue must be reported.

Reply:
I just update the code.
I resolve this conversation but feel free to reopen it if there is still something missing.